Framework/API/inc/MantidAPI/Algorithm.tcc

```diff
@@ -2,7 +2,9 @@
 #include "MantidAPI/IndexProperty.h"
 #include "MantidAPI/WorkspaceProperty.h"
 
+#ifdef _WIN32
 #include "MantidKernel/ArrayProperty.tcc"
```

Comment:
I have done something ugly here but this is currently the only way I can get the build to work on both windows and linux. Another pair of eyes on how I have defined `IndexProperty` and the methods in `Algorithm` for managing this property would be helpful. N.B. adding the definitions of `ArrayProperty` methods to the header opens a whole other can of worms and causes the build to fail.